code/game/machinery/suit_storage_unit.dm

```diff
@@ -332,6 +332,8 @@
 		I.play_tool_sound(user, I.tool_volume)
 
 /obj/machinery/suit_storage_unit/proc/store_item(obj/item/I, mob/user)
+	if(I.flags & NODROP)
```

Comment:
not the best way to handle it.

Reply:
If I check for drop_item early, I need to put it in each case, since it needs to happen before the `suit = I` assignment but after the check if it goes in the right slot (since we want to be sure there's a free slot before we drop it from the hand)

What I've done instead is use the `canunequip` proc, which is the same proc `drop_item` uses internally to check if it can actually drop the item. 

I hope that's an acceptable solution.